eth/downloader/downloader.go

```diff
@@ -1987,17 +1923,22 @@ func (d *Downloader) processFastSyncContent() error {
 			// Note, we have `reorgProtHeaderDelay` number of blocks withheld, Those
 			// need to be taken into account, otherwise we're detecting the pivot move
 			// late and will drop peers due to unavailable state!!!
-			if height := latest.Number.Uint64(); height > pivot.Number.Uint64()+2*max(d.epoch, fsMinFullBlocks)-uint64(reorgProtHeaderDelay) {
-				newPivot := d.calcPivot(height)
-				log.Warn("Pivot became stale, moving", "old", pivot, "new", newPivot)
+			// BUT we are not expecting chain reorgs in IBFT (so we could assume it as zero)
+			headerDelay := reorgProtHeaderDelay
+			if d.ibftConsensus {
+				headerDelay = 0
+			}
+			if height := latest.Number.Uint64(); height > pivot.Number.Uint64()+2*uint64(fsMinFullBlocks)-uint64(headerDelay) {
```

Comment:
So the `reorgProtHeaderDelay` was added in 6ee3b26f447459d3f3a316dbb572e461a273e193 and IIUC its intended use at the time was to prevent syncing headers at the very tip of the chain in the case that the sync head was not close to the tip, in order to avoid a situation where a node syncs a header that then gets re-orged. But here it appears that the `reorgProtHeaderDelay` is being used in a different way, looking at the above comment it seems that they are using `reorgProtHeaderDelay` to avoid reaching a point where the pivot point is stale and requests for state start failing. It seems that we would want this.

Reply:
Yes, agree on that, but in our case, check this line:
https://github.com/celo-org/celo-blockchain/pull/1958/files#diff-80cc409bf7f1e4a3d19174ed878e4dcb3d40b19b6e0580bd3816a174c57e0fb2R1227
We already removed that delay because we are IBFT.
So in this case it's maintaining the same logic, as there are no reorg, we removed that delay, so we shouldn' take that delay for this scenario

Reply:
Ok I've got it. Because eth was delaying pulling in `headerDelay` headers the sync head is `headerDelay` behind the real head and because state is pruned at 128 blocks after the real head we need to adjust our calculation of the prune point to take into account that the sync head is behind the real head.